File: unity-plugin/src/main/java/com/example/blehid/unity/events/DeviceEvent.java
package com.example.blehid.unity.events;

import android.bluetooth.BluetoothDevice;

/**
 * Base class for all device-related events.
 * Provides common device properties.
 */
public abstract class DeviceEvent extends BaseEvent {
    private final String deviceAddress;
    private final String deviceName;
    
    /**
     * Creates a new device event.
     * 
     * @param eventType Type of this event
     * @param device The Bluetooth device associated with the event
     */
    protected DeviceEvent(EventType eventType, BluetoothDevice device) {
        super(eventType);
        
        if (device == null) {
            this.deviceAddress = null;
            this.deviceName = null;
        } else {
            this.deviceAddress = device.getAddress();
            this.deviceName = device.getName();
        }
    }
    
    /**
     * Creates a new device event using just the address.
     * 
     * @param eventType Type of this event
     * @param deviceAddress MAC address of the device
     */
    protected DeviceEvent(EventType eventType, String deviceAddress) {
        super(eventType);
        this.deviceAddress = deviceAddress;
        this.deviceName = null; // Name unknown when only address is provided
    }
    
    /**
     * Gets the MAC address of the device.
     * 
     * @return Device MAC address
     */
    public String getDeviceAddress() {
        return deviceAddress;
    }
    
    /**
     * Gets the name of the device, if available.
     * 
     * @return Device name, or null if not available
     */
    public String getDeviceName() {
        return deviceName;
    }
    
    @Override
    public String toString() {
        return String.format("%s[deviceAddress=%s, deviceName=%s]", 
                             super.toString(), deviceAddress, deviceName);
    }
}

File: unity-test/Assets/BleHidTest/Events/DeviceEvent.cs
using System;

namespace BleHid.Events
{
    /// <summary>
    /// Base class for all device-related events.
    /// Provides common device properties.
    /// </summary>
    public abstract class DeviceEvent : BleHidEvent
    {
        /// <summary>
        /// MAC address of the device
        /// </summary>
        public string DeviceAddress { get; private set; }
        
        /// <summary>
        /// Name of the device, if available
        /// </summary>
        public string DeviceName { get; private set; }
        
        /// <summary>
        /// Creates a new device event.
        /// </summary>
        /// <param name="eventType">Type of this event</param>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="deviceName">Name of the device, if available</param>
        /// <param name="eventId">ID of the event (from Java side)</param>
        /// <param name="javaTimestamp">Java timestamp (milliseconds since epoch)</param>
        protected DeviceEvent(BleHidEventType eventType, string deviceAddress, string deviceName, Guid eventId, long javaTimestamp)
            : base(eventType, eventId, javaTimestamp)
        {
            DeviceAddress = deviceAddress;
            DeviceName = deviceName;
        }
        
        /// <summary>
        /// Creates a new device event.
        /// For events created on the Unity side.
        /// </summary>
        /// <param name="eventType">Type of this event</param>
        /// <param name="deviceAddress">MAC address of the device</param>
        /// <param name="deviceName">Name of the device, if available</param>
        protected DeviceEvent(BleHidEventType eventType, string deviceAddress, string deviceName = null)
            : base(eventType)
        {
            DeviceAddress = deviceAddress;
            DeviceName = deviceName;
        }
        
        /// <summary>
        /// Gets a string representation of this event.
        /// </summary>
        public override string ToString()
        {
            return $"{base.ToString()}, deviceAddress={DeviceAddress}, deviceName={DeviceName}";
        }
    }
}
